rules/import-style.js

```diff
@@ -127,6 +128,169 @@ const defaultStyles = {
 	},
 };
 
+const specialCases = {
+	react: 'React',
+	'react-dom': 'ReactDOM',
+	'react-router': 'ReactRouter',
+	'react-router-dom': 'ReactRouterDOM',
+	'prop-types': 'PropTypes',
+	lodash: '_',
+	'lodash-es': '_',
+	jquery: '$',
+	'styled-components': 'styled',
+	redux: 'Redux',
+	'react-redux': 'ReactRedux',
+	axios: 'Axios',
+	ramda: 'R',
+	rxjs: 'Rx',
+	vue: 'Vue',
+	angular: 'Angular',
+};
+
+const getNamespaceIdentifier = moduleName => {
+	// Handle special cases first
+	if (specialCases[moduleName]) {
+		return specialCases[moduleName];
+	}
+
+	// Trim trailing slashes and get the last part of the path and remove extension
+	const lastPart = moduleName.replace(/\/+$/, '').split('/').pop().split('.')[0];
+
+	// Replace invalid identifier characters and convert to camelCase
+	let identifier = lastPart
+		.replaceAll(/[^\dA-Za-z-]/g, '-') // Replace invalid chars with hyphen
+		.replaceAll(/-./g, x => x[1].toUpperCase()); // Convert to camelCase
+
+	// Prefix with underscore if starts with a number
+	if (/^\d/.test(identifier)) {
+		identifier = '_' + identifier;
+	}
+
+	return identifier;
+};
+
+/**
+ Creates a fix generator function to convert imports to namespace imports.
+ @param {Object} options - The options object.
+ @param {import('estree').Node} options.node - The AST node representing the import/require statement.
+ @param {import('eslint').SourceCode} options.sourceCode - The ESLint source code object.
+ @param {string} options.moduleName - The name of the module being imported.
+ @returns {(fixer: import('eslint').Rule.RuleFixer) => Generator<import('eslint').Rule.Fix, void, undefined> | undefined} A function that takes a fixer and returns a generator of fixes.
+ */
+const fix = ({node, sourceCode, moduleName, allowedImportStyles}) => {
+	// Currently we only fix named to namespace
+	if (!allowedImportStyles.has('namespace') || allowedImportStyles.has('named')) {
+		return;
+	}
+
+	const isImportDeclaration = node.type === 'ImportDeclaration';
+	const isVariableDeclarator = node.type === 'VariableDeclarator';
+	const isRequireCall = isCallExpression(node.init, {name: 'require'});
+	const isImportExpression = node.init?.type === 'ImportExpression';
+	const isVariableDeclaratorWithRequireCall = isVariableDeclarator && (isRequireCall || isImportExpression);
+	const isValidNode = isImportDeclaration || isVariableDeclaratorWithRequireCall;
+
+	if (!isValidNode) {
+		return;
+	}
+
+	let importedNames = [];
+	if (node.type === 'ImportDeclaration') {
+		importedNames = node.specifiers
+			.filter(specifier => specifier.type === 'ImportSpecifier' || specifier.type === 'ImportDefaultSpecifier')
+			.map(specifier => ({
+				localName: specifier.local.name,
+				importedName: specifier.type === 'ImportDefaultSpecifier' ? 'default' : specifier.imported.name,
+			}));
+	} else if (node.type === 'VariableDeclarator' && node.id.type === 'ObjectPattern') {
```

Comment:
I think we can focus on `import`s, leave `require()` alone, ESM should be preferred.

Reply:
Yes, we no longer plan to handle anything CommonJS.